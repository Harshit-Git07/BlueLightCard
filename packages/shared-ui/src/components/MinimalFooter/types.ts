import { PropsWithChildren } from 'react';

export interface NavItem {
  text: string;
  link: string;
}

export type MinimalFooterProps = PropsWithChildren & {
  navItems: NavItem[];
};
