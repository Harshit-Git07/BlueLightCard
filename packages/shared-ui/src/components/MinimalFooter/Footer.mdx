import { Meta, Canvas, Controls } from '@storybook/blocks';

import * as Footer from './Footer.stories';

<Meta of={Footer} />

<Canvas of={Footer.Default} />
