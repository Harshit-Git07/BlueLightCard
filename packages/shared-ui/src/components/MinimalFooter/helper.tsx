import { env } from '../../env';
export const getCopyrightText = () => {
  const currentYear = new Date().getFullYear();

  return env.APP_BRAND === 'dds-uk' ? (
    <>
      <span className="block">{`© Defence Discount Service 2012 - ${currentYear}`}</span>
      <span className="block">Operated by Blue Light Card Ltd</span>
    </>
  ) : (
    `©Blue Light Card 2008 - ${currentYear}`
  );
};

const blcUkNavItems = [
  {
    text: 'Terms & Conditions',
    link: 'https://www.bluelightcard.co.uk/terms_and_conditions.php',
  },
  {
    text: 'Privacy Notice',
    link: 'https://www.bluelightcard.co.uk/privacy-notice.php',
  },
  {
    text: 'Candidate Privacy Notice',
    link: 'https://www.bluelightcard.co.uk/candidate-privacy-notice.php',
  },
  {
    text: 'Cookie Policy',
    link: 'https://www.bluelightcard.co.uk/cookies_policy.php',
  },
  {
    text: 'Legal and Regulatory',
    link: 'https://www.bluelightcard.co.uk/legal-and-regulatory.php',
  },
];

const blcAuNavItems = [
  {
    text: 'Terms & Conditions',
    link: 'https://www.bluelightcard.com.au/terms_and_conditions.php',
  },
  {
    text: 'Privacy Notice',
    link: 'https://www.bluelightcard.com.au/privacy-notice.php',
  },
  {
    text: 'Candidate Privacy Notice',
    link: 'https://www.bluelightcard.com.au/candidate-privacy-notice.php',
  },
  {
    text: 'Cookie Policy',
    link: 'https://www.bluelightcard.com.au/cookies_policy.php',
  },
];

const ddsUkNavItems = [
  {
    text: 'Terms & Conditions',
    link: 'https://www.defencediscountservice.co.uk/terms_and_conditions.php',
  },
  {
    text: 'Privacy Notice',
    link: 'https://www.defencediscountservice.co.uk/privacy-notice.php',
  },
  {
    text: 'Cookie Policy',
    link: 'https://www.defencediscountservice.co.uk/cookies_policy.php',
  },
  {
    text: 'Legal and Regulatory',
    link: 'https://www.defencediscountservice.co.uk/legal-and-regulatory.php',
  },
];

export const getNavItems = () => {
  const brand = env.APP_BRAND;
  switch (brand) {
    case 'dds-uk':
      return ddsUkNavItems;
    case 'blc-au':
      return blcAuNavItems;
    default:
      return blcUkNavItems;
  }
};
