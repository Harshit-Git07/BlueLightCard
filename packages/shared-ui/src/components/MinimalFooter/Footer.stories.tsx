import { Meta, StoryFn } from '@storybook/react';
import MinimalFooter from '.';

const meta: Meta<typeof MinimalFooter> = {
  title: 'Component System/Minimal Footer',
  component: MinimalFooter,
};

const FooterTemplate: StoryFn<typeof MinimalFooter> = (args) => <MinimalFooter />;
export const Default = FooterTemplate.bind({});
Default.args = {};

export default meta;
