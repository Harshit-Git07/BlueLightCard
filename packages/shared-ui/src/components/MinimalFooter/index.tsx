import * as React from 'react';
import Link from 'next/link';
import { getCopyrightText, getNavigationItems } from './helper';

const MinimalFooter = () => {
  const copyrightText = getCopyrightText(),
    navigationItems = getNavigationItems();

  return (
    <div className="w-full max-w-[374px] flex flex-col items-center gap-4 px-4 pb-4 md:max-w-full md:px-4 lg:px-[56px] lg:flex-row lg:justify-between lg:items-center lg:max-w-[1728px] bg-colour-surface dark:bg-colour-surface-dark">
      <div className="w-fit border-t-[1px] border-colour-onSurface-outline dark:border-colour-onSurface-outline-dark flex flex-col lg:flex-row lg:justify-between items-center w-full">
        <p
          className="mt-6 text-center text-colour-onSurface-subtle-light dark:text-colour-onSurface-subtle-dark text-base font-typography-body-light font-typography-body-light-weight tracking-typography-body-light leading-typography-body-light leading-7 tracking-tight whitespace-nowrap max-w-full"
          role="contentinfo"
          aria-label="Copyright Information"
        >
          {copyrightText}
        </p>
        <div className="grid grid-cols-2 gap-4 justify-items-center w-full text-center md:grid-cols-none md:grid-rows-none md:flex md:flex-row md:justify-center lg:justify-end lg:gap-[16px]">
          {navigationItems.map((item, index) => (
            <Link
              id={item.text + '-nav-item'}
              aria-label={item.text + ' footer link'}
              key={item.text}
              href={item.link}
              className={`mt-6 text-colour-onSurface-subtle-light dark:text-colour-onSurface-subtle-dark text-base font-typography-body-light font-typography-body-light-weight tracking-typography-body-light leading-typography-body-light ${
                navigationItems.length % 2 !== 0 && index === navigationItems.length - 1
                  ? 'col-span-2 justify-self-center md:col-span-1'
                  : 'w-full md:w-auto'
              }`}
            >
              {item.text}
            </Link>
          ))}
        </div>
      </div>
    </div>
  );
};

export default MinimalFooter;
