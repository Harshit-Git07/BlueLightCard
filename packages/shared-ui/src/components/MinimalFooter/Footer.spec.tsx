import MinimalFooter from './index';
import renderer from 'react-test-renderer';

describe('MinimalFooter', () => {
  describe('snapshot Test', () => {
    it('renders a blc-uk footer', () => {
      const component = renderer.create(<MinimalFooter />);
      const tree = component.toJSON();
      expect(tree).toMatchSnapshot();
    });
  });

  it('renders a blc-au footer', () => {
    process.env.NEXT_PUBLIC_APP_BRAND = 'blc-au';
    const component = renderer.create(<MinimalFooter />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders a dds-uk footer', () => {
    process.env.NEXT_PUBLIC_APP_BRAND = 'dds-uk';
    const component = renderer.create(<MinimalFooter />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
