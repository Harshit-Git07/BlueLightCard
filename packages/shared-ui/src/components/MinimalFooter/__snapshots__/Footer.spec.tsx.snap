// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MinimalFooter renders a blc-au footer 1`] = `
<div
  className="w-full max-w-[374px] flex flex-col items-center gap-4 px-4 pb-4 md:max-w-full md:px-4 lg:px-[56px] lg:flex-row lg:justify-between lg:items-center lg:max-w-[1728px] bg-colour-surface dark:bg-colour-surface-dark"
>
  <div
    className="w-fit border-t-[1px] border-colour-onSurface-outline dark:border-colour-onSurface-outline-dark flex flex-col lg:flex-row lg:justify-between items-center w-full"
  >
    <p
      className="mt-6 text-center text-colour-onSurface-subtle-light dark:text-colour-onSurface-subtle-dark text-lg font-typography-body-light font-typography-body-light-weight tracking-typography-body-light leading-typography-body-light leading-7 tracking-tight whitespace-nowrap max-w-full"
    >
      ©Blue Light Card 2008 - 2024
    </p>
    <div
      className="grid grid-cols-2 gap-4 justify-items-center w-full text-center md:grid-cols-none md:grid-rows-none md:flex md:flex-row md:justify-center lg:justify-end lg:gap-[16px]"
    >
      <a
        aria-label="Terms & Conditions footer link"
        className="mt-6 text-colour-onSurface-subtle-light dark:text-colour-onSurface-subtle-dark font-typography-body-light font-typography-body-light-weight tracking-typography-body-light leading-typography-body-light w-full md:w-auto"
        href="https://www.bluelightcard.co.uk/terms_and_conditions.php"
        id="Terms & Conditions-nav-item"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onTouchStart={[Function]}
      >
        Terms & Conditions
      </a>
      <a
        aria-label="Privacy Notice footer link"
        className="mt-6 text-colour-onSurface-subtle-light dark:text-colour-onSurface-subtle-dark font-typography-body-light font-typography-body-light-weight tracking-typography-body-light leading-typography-body-light w-full md:w-auto"
        href="https://www.bluelightcard.co.uk/privacy-notice.php"
        id="Privacy Notice-nav-item"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onTouchStart={[Function]}
      >
        Privacy Notice
      </a>
      <a
        aria-label="Candidate Privacy Notice footer link"
        className="mt-6 text-colour-onSurface-subtle-light dark:text-colour-onSurface-subtle-dark font-typography-body-light font-typography-body-light-weight tracking-typography-body-light leading-typography-body-light w-full md:w-auto"
        href="https://www.bluelightcard.co.uk/candidate-privacy-notice.php"
        id="Candidate Privacy Notice-nav-item"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onTouchStart={[Function]}
      >
        Candidate Privacy Notice
      </a>
      <a
        aria-label="Cookie Policy footer link"
        className="mt-6 text-colour-onSurface-subtle-light dark:text-colour-onSurface-subtle-dark font-typography-body-light font-typography-body-light-weight tracking-typography-body-light leading-typography-body-light w-full md:w-auto"
        href="https://www.bluelightcard.co.uk/cookies_policy.php"
        id="Cookie Policy-nav-item"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onTouchStart={[Function]}
      >
        Cookie Policy
      </a>
      <a
        aria-label="Legal and Regulatory footer link"
        className="mt-6 text-colour-onSurface-subtle-light dark:text-colour-onSurface-subtle-dark font-typography-body-light font-typography-body-light-weight tracking-typography-body-light leading-typography-body-light col-span-2 justify-self-center md:col-span-1"
        href="https://www.bluelightcard.co.uk/legal-and-regulatory.php"
        id="Legal and Regulatory-nav-item"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onTouchStart={[Function]}
      >
        Legal and Regulatory
      </a>
    </div>
  </div>
</div>
`;

exports[`MinimalFooter renders a dds-uk footer 1`] = `
<div
  className="w-full max-w-[374px] flex flex-col items-center gap-4 px-4 pb-4 md:max-w-full md:px-4 lg:px-[56px] lg:flex-row lg:justify-between lg:items-center lg:max-w-[1728px] bg-colour-surface dark:bg-colour-surface-dark"
>
  <div
    className="w-fit border-t-[1px] border-colour-onSurface-outline dark:border-colour-onSurface-outline-dark flex flex-col lg:flex-row lg:justify-between items-center w-full"
  >
    <p
      className="mt-6 text-center text-colour-onSurface-subtle-light dark:text-colour-onSurface-subtle-dark text-lg font-typography-body-light font-typography-body-light-weight tracking-typography-body-light leading-typography-body-light leading-7 tracking-tight whitespace-nowrap max-w-full"
    >
      ©Blue Light Card 2008 - 2024
    </p>
    <div
      className="grid grid-cols-2 gap-4 justify-items-center w-full text-center md:grid-cols-none md:grid-rows-none md:flex md:flex-row md:justify-center lg:justify-end lg:gap-[16px]"
    >
      <a
        aria-label="Terms & Conditions footer link"
        className="mt-6 text-colour-onSurface-subtle-light dark:text-colour-onSurface-subtle-dark font-typography-body-light font-typography-body-light-weight tracking-typography-body-light leading-typography-body-light w-full md:w-auto"
        href="https://www.bluelightcard.co.uk/terms_and_conditions.php"
        id="Terms & Conditions-nav-item"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onTouchStart={[Function]}
      >
        Terms & Conditions
      </a>
      <a
        aria-label="Privacy Notice footer link"
        className="mt-6 text-colour-onSurface-subtle-light dark:text-colour-onSurface-subtle-dark font-typography-body-light font-typography-body-light-weight tracking-typography-body-light leading-typography-body-light w-full md:w-auto"
        href="https://www.bluelightcard.co.uk/privacy-notice.php"
        id="Privacy Notice-nav-item"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onTouchStart={[Function]}
      >
        Privacy Notice
      </a>
      <a
        aria-label="Candidate Privacy Notice footer link"
        className="mt-6 text-colour-onSurface-subtle-light dark:text-colour-onSurface-subtle-dark font-typography-body-light font-typography-body-light-weight tracking-typography-body-light leading-typography-body-light w-full md:w-auto"
        href="https://www.bluelightcard.co.uk/candidate-privacy-notice.php"
        id="Candidate Privacy Notice-nav-item"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onTouchStart={[Function]}
      >
        Candidate Privacy Notice
      </a>
      <a
        aria-label="Cookie Policy footer link"
        className="mt-6 text-colour-onSurface-subtle-light dark:text-colour-onSurface-subtle-dark font-typography-body-light font-typography-body-light-weight tracking-typography-body-light leading-typography-body-light w-full md:w-auto"
        href="https://www.bluelightcard.co.uk/cookies_policy.php"
        id="Cookie Policy-nav-item"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onTouchStart={[Function]}
      >
        Cookie Policy
      </a>
      <a
        aria-label="Legal and Regulatory footer link"
        className="mt-6 text-colour-onSurface-subtle-light dark:text-colour-onSurface-subtle-dark font-typography-body-light font-typography-body-light-weight tracking-typography-body-light leading-typography-body-light col-span-2 justify-self-center md:col-span-1"
        href="https://www.bluelightcard.co.uk/legal-and-regulatory.php"
        id="Legal and Regulatory-nav-item"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onTouchStart={[Function]}
      >
        Legal and Regulatory
      </a>
    </div>
  </div>
</div>
`;

exports[`MinimalFooter snapshot Test renders a blc-uk footer 1`] = `
<div
  className="w-full max-w-[374px] flex flex-col items-center gap-4 px-4 pb-4 md:max-w-full md:px-4 lg:px-[56px] lg:flex-row lg:justify-between lg:items-center lg:max-w-[1728px] bg-colour-surface dark:bg-colour-surface-dark"
>
  <div
    className="w-fit border-t-[1px] border-colour-onSurface-outline dark:border-colour-onSurface-outline-dark flex flex-col lg:flex-row lg:justify-between items-center w-full"
  >
    <p
      className="mt-6 text-center text-colour-onSurface-subtle-light dark:text-colour-onSurface-subtle-dark text-lg font-typography-body-light font-typography-body-light-weight tracking-typography-body-light leading-typography-body-light leading-7 tracking-tight whitespace-nowrap max-w-full"
    >
      ©Blue Light Card 2008 - 2024
    </p>
    <div
      className="grid grid-cols-2 gap-4 justify-items-center w-full text-center md:grid-cols-none md:grid-rows-none md:flex md:flex-row md:justify-center lg:justify-end lg:gap-[16px]"
    >
      <a
        aria-label="Terms & Conditions footer link"
        className="mt-6 text-colour-onSurface-subtle-light dark:text-colour-onSurface-subtle-dark font-typography-body-light font-typography-body-light-weight tracking-typography-body-light leading-typography-body-light w-full md:w-auto"
        href="https://www.bluelightcard.co.uk/terms_and_conditions.php"
        id="Terms & Conditions-nav-item"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onTouchStart={[Function]}
      >
        Terms & Conditions
      </a>
      <a
        aria-label="Privacy Notice footer link"
        className="mt-6 text-colour-onSurface-subtle-light dark:text-colour-onSurface-subtle-dark font-typography-body-light font-typography-body-light-weight tracking-typography-body-light leading-typography-body-light w-full md:w-auto"
        href="https://www.bluelightcard.co.uk/privacy-notice.php"
        id="Privacy Notice-nav-item"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onTouchStart={[Function]}
      >
        Privacy Notice
      </a>
      <a
        aria-label="Candidate Privacy Notice footer link"
        className="mt-6 text-colour-onSurface-subtle-light dark:text-colour-onSurface-subtle-dark font-typography-body-light font-typography-body-light-weight tracking-typography-body-light leading-typography-body-light w-full md:w-auto"
        href="https://www.bluelightcard.co.uk/candidate-privacy-notice.php"
        id="Candidate Privacy Notice-nav-item"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onTouchStart={[Function]}
      >
        Candidate Privacy Notice
      </a>
      <a
        aria-label="Cookie Policy footer link"
        className="mt-6 text-colour-onSurface-subtle-light dark:text-colour-onSurface-subtle-dark font-typography-body-light font-typography-body-light-weight tracking-typography-body-light leading-typography-body-light w-full md:w-auto"
        href="https://www.bluelightcard.co.uk/cookies_policy.php"
        id="Cookie Policy-nav-item"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onTouchStart={[Function]}
      >
        Cookie Policy
      </a>
      <a
        aria-label="Legal and Regulatory footer link"
        className="mt-6 text-colour-onSurface-subtle-light dark:text-colour-onSurface-subtle-dark font-typography-body-light font-typography-body-light-weight tracking-typography-body-light leading-typography-body-light col-span-2 justify-self-center md:col-span-1"
        href="https://www.bluelightcard.co.uk/legal-and-regulatory.php"
        id="Legal and Regulatory-nav-item"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onTouchStart={[Function]}
      >
        Legal and Regulatory
      </a>
    </div>
  </div>
</div>
`;
